Lib/typing.py

```diff
@@ -1786,8 +1786,9 @@ def __typing_unpacked_tuple_args__(self):
         assert self.__origin__ is Unpack
         assert len(self.__args__) == 1
         arg, = self.__args__
-        if isinstance(arg, _GenericAlias):
-            assert arg.__origin__ is tuple
+        if isinstance(arg, (_GenericAlias, types.GenericAlias)):
+            if arg.__origin__ is not tuple:
+                raise TypeError("Unpack[...] must be used with a tuple type")
```

Comment:
It feels weirdly inconsistent that we allow nonsensical things with the new syntax, but not with the "backwards-compatible syntax":

```pytb
>>> from typing import TypeVarTuple, Unpack
>>> class Foo[*Ts]: ...
...
>>> Ts = TypeVarTuple("Ts")
>>> Bar = Foo[int, *Ts]
>>> Bar[*list[str]]
__main__.Foo[int, *list[str]]
>>> Bar[Unpack[list[str]]]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
    Bar[Unpack[list[str]]]
    ~~~^^^^^^^^^^^^^^^^^^^
  File "/Users/alexw/dev/cpython/Lib/typing.py", line 394, in inner
    return func(*args, **kwds)
           ~~~~^^^^^^^^^^^^^^^
  File "/Users/alexw/dev/cpython/Lib/typing.py", line 1394, in __getitem__
    args = _unpack_args(args)
           ~~~~~~~~~~~~^^^^^^
  File "/Users/alexw/dev/cpython/Lib/typing.py", line 306, in _unpack_args
    subargs = getattr(arg, '__typing_unpacked_tuple_args__', None)
              ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alexw/dev/cpython/Lib/typing.py", line 1791, in __typing_unpacked_tuple_args__
    raise TypeError("Unpack[...] must be used with a tuple type")
TypeError: Unpack[...] must be used with a tuple type
```

But I guess that's a preexisting problem, not new to this PR